aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1606,17 +1606,38 @@ std::tuple<Tensor&, Tensor&> triangular_solve_out(const Tensor& self, const Tens
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ qr ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+/*
+  The geqrf function computes QR decomposition of matrices stored in `self`.
+  However, rather than producing a Q matrix directly, it produces a sequence of
+  elementary reflectors which may later be composed to construct Q - for example
+  with the orgqr or ormqr functions.
+
+  Args:
+  * `self` - [in] Input tensor for QR decomposition
+             [out] QR decomposition result which contains:
+              i)  The elements of R, on and above the diagonal.
+              ii) Directions of the reflectors implicitly defining Q.
+             Tensor with the directions of the elementary reflectors below the diagonal,
+              it will be overwritten with the result
+  * `tau` - [out] Tensor which will contain the magnitudes of the reflectors
+            implicitly defining Q.
+  * `m` - The number of rows of `self` to consider
+  * `n` - The number of columns of `self` to consider (actual sizes of `self` could be larger)
+
+  For further details, please see the LAPACK documentation for GEQRF.
+*/
 template <typename scalar_t>
-static void apply_geqrf(Tensor& self, Tensor& tau, int64_t m, int64_t n) {
+static void apply_geqrf(const Tensor& self, const Tensor& tau, int64_t m, int64_t n) {
 #ifndef USE_LAPACK
-  AT_ERROR("qr: LAPACK library not found in compilation");
+  AT_ERROR("geqrf: LAPACK library not found in compilation");
```

Comment:
TORCH_CHECK(false, ...)

Reply:
We have a different error string for LAPACK not being available in the other linalg ops, right?

Reply:
It's not consistent, some use the message as here and others use a nicer message. I'll change it here to a nicer message.